otx/cli/tools/build.py

```diff
@@ -57,56 +59,100 @@ def parse_args():
-        if args.task is None:
-            task_type = config_manager.auto_task_detection(args.train_data_roots)
-            args.task = task_type
-        if args.val_data_roots is None:
-            config_manager.auto_split_data(args.train_data_roots, args.task)
-
-    if args.task and args.task in SUPPORTED_TASKS:
+    if train_data_roots:
+        if task is None:
+            task_type = config_manager.auto_task_detection(train_data_roots)
+            task = task_type
+        if val_data_roots is None:
+            config_manager.auto_split_data(train_data_roots, task)
+
+    if task is not None and task in SUPPORTED_TASKS:
         # Build with task_type and create user workspace
-        if args.workspace_root is None:
-            args.workspace_root = set_workspace(args.task, args.model)
+        if workspace_root is None:
+            workspace_root = set_workspace(task, model)
+
+        train_type = train_type.lower() if train_type is not None else train_type
         builder.build_task_config(
-            task_type=args.task,
-            model_type=args.model,
-            train_type=args.train_type.lower(),
-            workspace_path=Path(args.workspace_root),
+            task_type=task,
+            model_type=model,
+            train_type=train_type,
+            workspace_path=Path(workspace_root),
             otx_root=otx_root,
+            template=template,
         )
         config_manager.write_data_with_cfg(
-            workspace_dir=args.workspace_root,
-            train_data_roots=args.train_data_roots,
-            val_data_roots=args.val_data_roots,
+            workspace_dir=workspace_root,
+            train_data_roots=train_data_roots,
+            val_data_roots=val_data_roots,
         )
 
     # Build Backbone related
-    if args.backbone:
+    if backbone:
         missing_args = []
-        if not args.backbone.endswith((".yml", ".yaml", ".json")):
-            if args.save_backbone_to is None:
-                args.save_backbone_to = (
-                    os.path.join(args.workspace_root, "backbone.yaml") if args.workspace_root else "./backbone.yaml"
+        if not backbone.endswith((".yml", ".yaml", ".json")):
+            if save_backbone_to is None:
+                save_backbone_to = (
+                    os.path.join(workspace_root, "backbone.yaml") if workspace_root else "./backbone.yaml"
                 )
-            missing_args = builder.build_backbone_config(args.backbone, args.save_backbone_to)
-            args.backbone = args.save_backbone_to
-        if args.model:
+            missing_args = builder.build_backbone_config(backbone, save_backbone_to)
+            backbone = save_backbone_to
+        if model:
             if missing_args:
                 raise ValueError("The selected backbone has inputs that the user must enter.")
-            builder.merge_backbone(args.model, args.backbone)
+            builder.merge_backbone(model, backbone)
+
+
+def main():
```

Comment:
Do we need main functions? For what reasons? If we want to test that everything is working, maybe unit tests is a better choice?
Overwise, this main function also should be supported and it adds code to the file

Reply:
This is the CLI interface for `otx build`. So, if user gives arguments as `otx build --train-data-roots={}`, this main function will be executed.

BTW, @harimkang , @eunwoosh , and I discussed for the current class diagram especially the relationship between `Builder` and `ConfigManager`. As a results, we'll refactoring the overall class relationship and responsibility.

Reply:
@kprokofi This is what I agree to. The main function is actually the way all CLIs use it now, so we'll leave it there, but we're going to work on reducing the code in main function. Maybe one day we can get rid of it..?

Reply:
Hmm, if we remove the `main()` function, is there any way to support the CLI?

Reply:
In the `if __name__ == "__main__":` part, it would be nice to divide it into several functions with distinct functions other than main.

Reply:
Ah, sure, make sense